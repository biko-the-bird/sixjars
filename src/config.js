
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//this is a configuration for enzyme that needs
//to be imported in each testing file
const config = configure({ adapter: new Adapter() });
export default config;