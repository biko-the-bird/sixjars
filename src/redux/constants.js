export const SET_JARS = "SET_JARS";
export const WITHDRAW_JAR = "WITHDRAW_JAR";
export const JARS_COOKIE = "JARS_COOKIE";
export const NEC = "NEC";
export const FFA = "FFA";
export const EDU = "EDU";
export const GIV = "GIV";
export const LTS = "LTS";
export const PLA = "PLA";


export const INIT_JARS = {NEC: 0, FFA: 0,GIV: 0, PLA: 0,EDU: 0,LTS: 0};
export const JAR_DESC = {NEC: "Absolute neccesites only. i.e. Food, housing, transportation, clothes.", FFA: "Do not spend this account. You can only invest it.", GIV: "To be donated to things you want to support. i.e. homeless, buying a friends mixtape", PLA: "guilt free, spend on anything you want", EDU: "for classes or gym memberships.", LTS: "Savings for big purchases. i.e. a car"}