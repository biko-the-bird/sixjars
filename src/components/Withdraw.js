import React, {Component} from 'react';
import {connect} from 'react-redux';
import {withdrawJar} from '../redux/actions/balance';
import * as constants from '../redux/constants';

export class Withdraw extends Component {

    constructor(props) {
        super(props);
        this.state = {
            withdraw: 0,
            withdrawJar: constants.NEC
        }
    }

    handleWithdraw = () => {
        this.props.withdrawJar(this.state.withdrawJar, this.state.withdraw);
        this.setState({
            withdraw: 0
        });
    }

    render() {
        //console.log(this.props);
        const keys = Object.keys(this.props.jars);
        return (
            <div>

<label>Withdraw</label>

 <input
            type="number"
            className="withdraw-input"
            value={this.state.withdraw}
            onChange={e => this.setState({withdraw: e.target.value})}
          />
          <label>from</label>
             <select id="jar-selector" onChange={e => this.setState({withdrawJar: e.target.value})}>
          {keys.map(key => {
            return (
              <option value={key} key={key}>
                {key} 
              </option>
            );
          })}
        </select>
          <button className="withdraw-btn" onClick={() => this.handleWithdraw()}>
            Withdraw
          </button>
        
            </div>
        )
    }
}
const mapStateToProps = (state) => {
    return {jars: state};
}

export default connect( mapStateToProps,{withdrawJar})(Withdraw);