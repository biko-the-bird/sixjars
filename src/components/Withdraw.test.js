import React  from 'react';
import {shallow} from 'enzyme';
import { config } from "../config";
import {Withdraw} from './Withdraw';
import {INIT_JARS} from '../redux/constants';

describe("Withdraw", () => {
    const mockWithdrawJar = jest.fn();

    const props = {withdrawJar: mockWithdrawJar, jars: INIT_JARS};
    const withdraw = shallow(<Withdraw {...props}/>);
    it("renders properly", () => {
        expect(withdraw).toMatchSnapshot();
    });

    describe("user enters a new withdraw", () => {
        const input = 10;
        beforeEach(() => {
            withdraw.find('.withdraw-input').simulate('change', {target: {
                value: input
            }});
        });

        it("saves the entered value to local state", () => {
            expect(withdraw.state().withdraw).toEqual(input);
        });
    });

    describe("users submits a withdraw", () => {
        beforeEach(() => {
            withdraw.find('.withdraw-btn').simulate('click');
        });

        it("clears the local state", () => {
            expect(withdraw.state().withdraw).toEqual(0);
        });

        it("triggers `withdrawJar()` method from props", () => {
            expect(mockWithdrawJar).toHaveBeenCalled();
        })
    });
});