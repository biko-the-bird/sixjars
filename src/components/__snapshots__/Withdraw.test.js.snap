// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Withdraw renders properly 1`] = `ShallowWrapper {}`;
